verifier/src/evidence.rs

```diff
@@ -0,0 +1,334 @@
+        .expect("Failed to decode public key");
+
+        key
+    }
+
+    fn quote_signing_key<T: AsRef<[u8]>>(quote: &Quote3<T>) -> VerifyingKey {
+        let signature_data = quote.signature_data();
+        let certification_data = signature_data.certification_data();
+        let CertificationData::PckCertificateChain(chain) = certification_data else {
+            panic!("Unexpected certification data type");
+        };
+        let leaf_pem = chain.into_iter().next().expect("No leaf certificate");
+        let certificate =
+            Certificate::from_pem(leaf_pem).expect("Failed to parse leaf certificate");
+
+        let key = VerifyingKey::from_sec1_bytes(
+            certificate
+                .tbs_certificate
+                .subject_public_key_info
+                .subject_public_key
+                .as_bytes()
+                .expect("Failed to parse public key"),
+        )
+        .expect("Failed to decode public key");
+
+        key
+    }
+
+    type VerifierType<T> = And<
+        Quote3Verifier<T>,
+        And<MrSignerVerifier, And<AttributesVerifier, And<TcbInfoRawVerifier, AdvisoriesVerifier>>>,
+    >;
+
+    // Get a common verifier used in these tests.
+    //
+    // The values were taking from the on test files
+    // "../data/tests/hw_quote.dat" and
+    // "../data/tests/fmspc_00906ED50000_2023_05_10.json"
+    fn verifier<Q: AsRef<[u8]>>(product_id: IsvProductId, quote: &Quote3<Q>) -> VerifierType<Q> {
+        let quote_key = quote_signing_key(quote);
+        let tcb_key = tcb_signing_key();
+        let mr_signer = [
+            0xD4, 0x12, 0xA4, 0xF0, 0x7E, 0xF8, 0x38, 0x92, 0xA5, 0x91, 0x5F, 0xB2, 0xAB, 0x58,
+            0x4B, 0xE3, 0x1E, 0x18, 0x6E, 0x5A, 0x4F, 0x95, 0xAB, 0x5F, 0x69, 0x50, 0xFD, 0x4E,
+            0xB8, 0x69, 0x4D, 0x7B,
+        ];
+        let isv_svn = 0;
+        let attributes = Attributes::default()
+            .set_flags(7)
+            .set_extended_features_mask(7);
+        let time = "2023-06-08T13:43:27Z"
+            .parse::<DateTime>()
+            .expect("Failed to parse time");
+        let allowed_advisories = Advisories::new(
+            [
+                "INTEL-SA-00161",
+                "INTEL-SA-00219",
+                "INTEL-SA-00289",
+                "INTEL-SA-00334",
+                "INTEL-SA-00614",
+                "INTEL-SA-00615",
+                "INTEL-SA-00617",
+            ],
+            AdvisoryStatus::OutOfDate,
+        );
+        let verifier = And::new(
+            Quote3Verifier::new(quote_key),
+            And::new(
+                MrSignerVerifier::new(mr_signer.into(), product_id, isv_svn.into()),
+                And::new(
+                    AttributesVerifier::new(attributes),
+                    And::new(
+                        TcbInfoRawVerifier::new(tcb_key, time),
+                        AdvisoriesVerifier::new(allowed_advisories),
+                    ),
+                ),
+            ),
+        );
```

Comment:
Without the evidence type this would have needed to be 3 different verifiers, one for the TCB signature, one for the advisories, and another for all the quote stuff.
The manual extracting of the advisories and the 3 separate verifiers is what felt really clunky for a client.